src/adaXT/decision_tree/decision_tree.pyx

```diff
@@ -360,11 +360,17 @@ class DecisionTree(BaseModel):
                 decision_queue.append(cur_node.left_child)
                 decision_queue.append(cur_node.right_child)
 
+    def __check_sample_input(self, sample: ArrayLike, name: str, dtype: np.dtype) -> np.ndarray:
+        if sample.size > self.n_rows_predict:
+            raise ValueError(f"{name} has more elements than the given input\
+                             data X.")
+        return np.array(sample, dtype=dtype)
+
```

Comment:
We actually could use the same function for the `fit` method of both the DecisionTree and RandomForest. I would propose:
1) Move this method into the base_model
2) Add the logic with if ... is None into this method.
3) Replace  `__check_sample_weight` on DecisionTree in fit to this function as well and add it to RandomForest as well
4) Change the input of both fit methods to ArrayLike

Would that work?